pkg/tinyusb/hw/hw_sam0.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2022 Mesotic SAS
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_tinyusb
+ * @brief
+ * @{
+ *
+ * @brief       tinyUSB hardware driver for SAM0 MCUs
+ * @author      Dylan Laduranty <dylan.laduranty@mesotic.com>
+ */
+
+#include <errno.h>
+
+#include "periph_conf.h"
+#include "periph/gpio.h"
+#include "pm_layered.h"
+#include "log.h"
+#include "tusb.h"
+#include "device/usbd.h"
+#include "host/usbh.h"
+
+int tinyusb_hw_init(void)
+{
+    /* Initialize GPIO pins */
+    gpio_init(sam_usbdev_config[0].dp, GPIO_IN);
+    gpio_init(sam_usbdev_config[0].dm, GPIO_IN);
+    gpio_init_mux(sam_usbdev_config[0].dm, sam_usbdev_config[0].d_mux);
+    gpio_init_mux(sam_usbdev_config[0].dp, sam_usbdev_config[0].d_mux);
+
+    /* Initialize clocks */
+    sam0_gclk_enable(sam_usbdev_config[0].gclk_src);
```

Comment:
Where does `sam_usbdev_config` come from?

Reply:
This is the variable defined in each board `periph_conf.h`

Reply:
If the board has to define it, shouldn't it be part of this PR then?

Reply:
It is already defined actually. For all SAM0-based boards w/ `periph_usbdev` support.
I am just reusing existing information to easily setup the USB clock.
This is the same behavior as hw_stm32.c, I just did bother to wrap it nicely as @gschorcht since SAM0 only has one controller AFAIK.